### ISSUE TITLE
Write user documentation

### ISSUE THREAD

**Issue by nohal** [nohal/chartdldr_pi#44]
With focus on 2 user groups
- A total novice with no previous knowledge of anything
- A user already having charts and _thinking he knows what he's doing_

Both need pictures, preferably taken on Windows

**Comment by rgleason** [comment]
Pavel, I can get started on this if you'd like.  These are two good cases!
Should they be kind of separate with the novice first where we walk through entire sequence?
Then deal with user with charts and how to move them into the recommended structure and some of the FAQ and hangups.

**Comment by nohal** [comment]
Yes, let's start in https://github.com/nohal/chartdldr_pi/wiki - I do not want to include it in the official wiki until it is ready for release

**Comment by rgleason** [comment]
OK, I was just going to start in the wiki... how do I use this one?
Is is just like code...
 I create my own local repository...
and make html pages with screen shots.
Then I commit to you?
This should be a good exercise in proper git etiquette!
Is this how you want me to try to proceed?

--Humm this is the same repos as the plugin... where do I put the wiki?  Now I see .../wiki

**Comment by rgleason** [comment]
Ok I git clone https://github.com/nohal/chartdldr_pi.wiki.git
It made  chartdldr_pi.wiki directory alongsiide chartdldr_pi
in my o-plugin directory.

I'm going to start with screenshots and then fill in.

**Comment by nohal** [comment]
Leave the screenshots for later, there still might be changes to the GUI.
If it suits you better, you can edit the wiki directly on the web, no need to do it locally.

**Comment by nohal** [comment]
And do not worry much about the formatting, we need the text...

**Comment by rgleason** [comment]
Pavel,
Tomorrow morning I'll get my git commands straight and request a pull.
The screenshots help organize, and focus the process. I can do more as needed.
The css is a little rough, but it will be overwritten.

**Comment by nohal** [comment]
You can (and should) really use the editor online, there is absolutely no need to suffer with the syntax editing it locally...

**Comment by rgleason** [comment]
I've got first draft of new user largely done with screenshots and text in html format but this fails.

git checkout -b fcg-add
git push -u origin fcg-add

C:\Data-Dart\Up-Soft\Navigation\o-plugin\chartdldr_pi.wiki>git branch -va
- fcg-add               5a9c36b Created User Documentation for existing OpenCPN  user (markdown)
  master                5a9c36b Created User Documentation for existing OpenCPN  user (markdown)
  remotes/origin/HEAD   -> origin/master
  remotes/origin/master 5a9c36b Created User Documentation for existing OpenCPN  user (markdown)

C:\Data-Dart\Up-Soft\Navigation\o-plugin\chartdldr_pi.wiki>git push -u origin fcg-add
Username for 'https://github.com': rgleason
Password for 'https://rgleason@github.com':
error: The requested URL returned error: 403 while accessing https://github.com/
nohal/chartdldr_pi.wiki.git/info/refs
fatal: HTTP request failed

--- maybe it was my password.  --Nope I guess not.

**Comment by rgleason** [comment]
I can put the files up on OpenCPN Beta File Thingie and you can download, but I'd rather do it with git. What am I doing wrong here?

Do I need to clone your repository to mine
Then create a link of some sort to upload to my repos somehow? How? Then do a pull request?

**Comment by nohal** [comment]
You are trying to push into my repository. And it fails as you don't have write access there. It will really be much easier to do it the way you suggest.

**Comment by rgleason** [comment]
Can you pull 
 https://github.com/rgleason/chartdldr_pi.wiki.git

There is no way to push a wiki that I have found. It comes up saying that there are no changes.

It is just text, but there is html code in it with links to 13 screenshots.
can I upload them to the wiki somehow?

Otherwise I will put it up on one of my websites. In fact I'll do it now.

**Comment by rgleason** [comment]
Temporarily it is available here.
http://tgp-architects.com/User-Documentation-New.htm

**Comment by nohal** [comment]
Many thanks, it looks promising.
I have cleaned up that HTML mess in https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-new-OpenCPN-user so it now has the correct format and is just missing the images which you probably forgot to add to the repository.
Just for the future, please don't use any HTML tags in the pages - it is double work to later get rid of them.

**Comment by rgleason** [comment]
I find the html useful to link the images and write towards those images once in sequence....
I don't find it a mess.  
So tell me how to get the image up there please, it seems to remain a secret. --
Also how do I create links to the wiki text?  - Later - You've done it.
Then how do I push to your wiki?

So next I need to figure out how to upload and where, maybe it is not necessary.
http://tgp-architects.com/User-Documentation-New.htm

**Comment by rgleason** [comment]
Ok, I found a markdown image link and took that format for an http and then realized I could just paste
http://tgp-architects.com/ in front of the image names,  so now it is working.

If you want it uploaded somehow, please advise how,

**Comment by nohal** [comment]
The easiest way for you is to edit it directly, just click the Edit button on the top of the wiki page
![screenshot from 2015-01-28 15 08 06](https://cloud.githubusercontent.com/assets/249856/5947015/a1ada13c-a6ff-11e4-8860-c24ae13c75a1.png)
The syntax for a link is almost the same as for image, just without the exclamation mark at the beginning:

```
[New OpenCPN user](User-Documentation-for-new-OpenCPN-user)
```

results in
[New OpenCPN user](User-Documentation-for-new-OpenCPN-user)

For more help on the formatting, look at https://guides.github.com/features/mastering-markdown/ or when editing a page, simply click on the Help icon:
![screenshot from 2015-01-28 15 12 33](https://cloud.githubusercontent.com/assets/249856/5947092/529c4246-a700-11e4-99ef-a09b7d0b4f8e.png)

HTML is a mess, whatever you find it, we need to be able to use the data in other places (online help, PDF, etc.), that's why we need it to be convertible to other formats, HTML is simply just one of them.

**Comment by rgleason** [comment]
Pavel, 
I've started to get the hang of it.  I've edited 2 dozen times and added links to my website images.
Next I've made a new directory
rgleason/chartdldr_pi/images/
and I am going to git clone my repository and add the images and then push.
Then I can link the images to the text with a local hyperlink.

Is this on track  now?
Thanks for the help..

Check out the wiki now!

**Comment by nohal** [comment]
Yes, exactly what we need. Thank you!

**Comment by rgleason** [comment]
Having some troubles pushing to my rgleason/chartdldr_pi 
I have added the files to local   r-chartdldr_pi/images/
Refusing to git push 
Its got some ssh problem with my public key.  

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git commit -m "Add image
s for Wiki"
[fcg-add fc4b62c] Add images for Wiki
warning: LF will be replaced by CRLF in images/User-Documentation-New.htm.
The file will have its original line endings in your working directory.
warning: LF will be replaced by CRLF in images/User-Documentation-for-existing-O
penCPN-user.md.
The file will have its original line endings in your working directory.
 17 files changed, 143 insertions(+)
 create mode 100644 images/1-Chart-Downloader-Tab.png
 create mode 100644 images/10-ENC-Add-New-Predefined-Chart-Source.png
 create mode 100644 images/11-ENC-Chart-List-Updated-Charts-checked.png
 create mode 100644 images/12-Download-Selected-Charts.png
 create mode 100644 images/13-Add-New-Custom-Catalog.png
 create mode 100644 images/2-Add-New-Predefined-Chart-Source.png
 create mode 100644 images/3-OK-Predefined-Chart-Directory-Set-Automatically.png

 create mode 100644 images/4-OK-Predefined-Chart-Dir-Set-Automatically-frame.png

 create mode 100644 images/5-Catalog-Entered.png
 create mode 100644 images/6-Chart-List-Updated-Charts-checked.png
 create mode 100644 images/7-Downloading-Charts.png
 create mode 100644 images/8-Downloading-Charts.png
 create mode 100644 images/9-Three-charts-downloaded-with-show-local-files.png
 create mode 100644 images/Home.md
 create mode 100644 images/User-Documentation-New.htm
 create mode 100644 images/User-Documentation-for-existing-OpenCPN-user.md
 create mode 100644 images/User-Documentation-for-new-OpenCPN-user.md

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git branch -va
- fcg-add                      fc4b62c Add images for Wiki
  master                       61e5029 Create Readme
  remotes/origin/HEAD          -> origin/master
  remotes/origin/coverity_scan 0e8508f Reorganize dialog to give more visual spa
  ce to input fields and disable the combo on editing an existing source.
  remotes/origin/master        61e5029 Create Readme

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git remote add origin ht
tps://github.com/rgleason/chartdlr_pi.git
fatal: remote origin already exists.

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git remote -v
origin  git://github.com/rgleason/chartdldr_pi.git (fetch)
origin  git://github.com/rgleason/chartdldr_pi.git (push)

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git push
fatal: remote error:
  You can't push to git://github.com/rgleason/chartdldr_pi.git
  Use https://github.com/rgleason/chartdldr_pi.git

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git remote set-url origi
n git@github.com:rgleason/chartdldr_pi.git

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>git push
The authenticity of host 'github.com (192.30.252.129)' can't be established.
RSA key fingerprint is xxxxxx  deleted
Are you sure you want to continue connecting (yes/no)? y
Please type 'yes' or 'no': yes
Warning: Permanently added 'github.com,192.30.252.129' (RSA) to the list of know
n hosts.
Permission denied (publickey).
fatal: The remote end hung up unexpectedly

C:\Data-Dart\Up-Soft\Navigation\o-plugin\r-chartdldr_pi>

**Comment by rgleason** [comment]
Git config

[core]
    repositoryformatversion = 0
    filemode = false
    bare = false
    logallrefupdates = true
    symlinks = false
    ignorecase = true
    hideDotFiles = dotGitOnly
[remote "origin"]
    fetch = +refs/heads/_:refs/remotes/origin/_
    url = git@github.com:rgleason/chartdldr_pi.git
[branch "master"]
    remote = origin
    merge = refs/heads/master

**Comment by nohal** [comment]
It seems you have the remote and SSH git setup all wrong (most likely caused by executing `git remote set-url origin git@github.com:rgleason/chartdldr_pi.git`, why did you do it?). And it actually looks like the git config you posted does not belong to the wiki repository at all, if you are sure it does, change the line 
`url = git@github.com:rgleason/chartdldr_pi.git`
to
`url = https://github.com/rgleason/chartdldr_pi.wiki.git`
but better than anything, stop suffering with the local edits and git and do the text editing online, we can add the images at any time.

**Comment by rgleason** [comment]
Pavel, I decided to start over and started using Github for Windows more seriously, I thought it was a toy.  Using Github for Win, cloned  rgleason/chartdldr  then added the 14 image files and made a new branch Add-Wiki-Images and committed the 14 files to that branch. Then highliighted the New Branch on the left and "Synced" with my github repos.  Then went to my github repository and it was quite easy to get on that branch and make a pull request to nohal/chartdldr.  

Then I figured out how to make the new local links to images for Wiki Chartdldr and cut and pasted those into the New User Document which is working now.  EG:
https://github.com/rgleason/chartdldr_pi/blob/Add-Wiki-Images/images/1-Chart-Downloader-Tab.png

Now I notice that 
https://github.com/nohal/chartdldr_pi/wiki/User-Documentation-for-new-OpenCPN-user
Does not have the same stuff as
https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-new-OpenCPN-user

Now I wonder, How do I make a pull for the Wiki docs for New and Existing Users? So your github repository has the wiki docs?

**Comment by rgleason** [comment]
Could not find how I issued a Pull for the wiki pages.
Pavel, I like this markup utility, it works well. Nice and simple, quick, light and NOT cumbersome. Any chance of getting Opencpn on it?

**Comment by nohal** [comment]
Not any soon

**Comment by rgleason** [comment]
Could the wiki and images be put on a Sub-Wiki on github.com/OpenCPN Plugins/chartdldr_pi?
When done?

**Comment by rgleason** [comment]
It got changed to almost the line you suggested!
    url = https://github.com/rgleason/chartdldr_pi.git
Would just do the wiki docs
    url = https://github.com/rgleason/chartdldr_pi.wiki.git
Now I think the problem started by git clone and not issuing a git init?

[core]
    repositoryformatversion = 0
    filemode = false
    bare = false
    logallrefupdates = true
    symlinks = false
    ignorecase = true
    hideDotFiles = dotGitOnly
[remote "origin"]
    url = https://github.com/rgleason/chartdldr_pi.git
    fetch = +refs/heads/_:refs/remotes/origin/_
[branch "master"]
    remote = origin
    merge = refs/heads/master
[remote "upstream"]
    url = https://github.com/nohal/chartdldr_pi.git
    fetch = +refs/heads/_:refs/remotes/upstream/_
[branch "coverity_scan"]
    remote = origin
    merge = refs/heads/coverity_scan
[branch "Add-Wiki-Images"]
    remote = origin
    merge = refs/heads/Add-Wiki-Images

**Comment by nohal** [comment]
This is wrong, you need to point it to the wiki repository.

**Comment by rgleason** [comment]
Ok but I need the images too and they are in chartdlr_pi/images
How do I handle both?

Sorry I sent another pull request tryiing to get my 2 images Add-2-Wiki-Images. onto my repos

**Comment by rgleason** [comment]
Added two images 14-Chart-File  and 14-Chart-Group and created a description for the "Other steps needed to use Charts, referring to main OpenCPN documentation too.

**Comment by rgleason** [comment]
Next Starting Existing Users.

**Comment by rgleason** [comment]
Using Github for Windows, can't clone chartdlr_pi.wiki because that does not exist in my repository, I only have chartdldr_pi.  I can however do that manually. with git.

**Comment by rgleason** [comment]
Have roughed out Existing User.
Any suggestions?
https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-existing-OpenCPN-user
Nite..

**Comment by rgleason** [comment]
RE: This is wrong, point to wiki.
I'll be busy until Tuesday, but wil try to get this worked out.
None of this is very apparent or intuitive at this point.
For now the images are in MY repository which is better than nothing.
At some point I'll be able to move them when I figure out this wiki thing.
There do not appear to be any Help docs on this aspect.

**Comment by rgleason** [comment]
Have tried to do this. 2 hours. No success. Sorry.
Cloned  rgleason/chartdlder_pi.wiki
Put images in there on my desktop.
Made branch "add-images"
Added all the files to that branch, including  /images/ subfolder also with the images.
Commited
After changing config to use https rather than git  (by hand using a previous on that I knew worked)
Pushed  and it seemed to work.
Then went back to github.com/rgleason/chartdldr_pi  and clicked on Wiki
Saw no way to get to chartdldr_pi.wiki  directory.

Then tried to find a way to upload the images to the wiki one by one, found no such way.

Without speciffic instructions or tutorial this will go no further. It is a waste of time and has taken me 4 x longer than just doing it the way I would normally.

**Comment by rgleason** [comment]
Have found this
https://help.github.com/articles/adding-images-to-wikis/
not much help. Have already done this

**Comment by rgleason** [comment]
Do you want me to update screenshots?
Also need to improve text about
1. Add Catalog  "Predefined" and  "Custom"  how it works.
2. Use of Chart Directories, recursive, work at the subdirectory level needed for flexibility. Don't dupliicate directories above as it increases number of scans of db. etc.

Should we make a page in the Wiki yet?

**Comment by rgleason** [comment]
Dan,
I have updated the images as you suggested. It appears that the modified text and images are running a bit ahead of themselves, please see the wiki now.  It appears that there are now original items below that are more step by step,  Since you appear to have deleted the images I had originally in the first part, perhaps I have these images in the wrong order or something?

Any way I am not going back to linking to my website as it is much much easier. You can now just move the images where you intend.

I will try to get all the rest of the images back to links refering to my website/chartdldr  -- See the way the links are created in Edit mode.  Please try not to delete these as I get lost between what you want changed and what I had.

Thanks.

**Comment by rgleason** [comment]
Some of the chart catalogs  xml reside here:
https://github.com/chartcatalogs/catalogs

Yet they are not being seen when Updated for some reason....Failed to download
The URL is shown as 
 https://raw.githubusercontent.com/chartcatalogs/catalogs/master/CZ_IENC_Catalog.xml

When I try this in the URL
https://github.com/chartcatalogs/catalogs/CZ_IENC_Catalog.xml
it accepts the URL but when update is hit 
I get failed to download, make sure there is a working internet connection.

I have also tried to add a new catalog.

The only thing I have done is use Edit Catalog  to change the 
 Chart Directory:  C:\Users\All Users\opencpn\charts\ENC\CZ_INLAND
from  C:\ProgramData\opencpn\Charts\ENC\CZ_INLAND

Maybe the "C" in charts makes a difference.....  Nope sitll fails.

Then tried adding a new Czech Chart Catalog, It shows the same as the new path that I tried above.
but when "Update" to get the chart list, it fails too.

TRied closing the plugin, then Opencpn and reopened and get the same problem.
![opencpn-2-27-2015 ini](https://cloud.githubusercontent.com/assets/2363959/6423675/9ca16d8e-beb7-11e4-866e-d0bfca5a5957.png)
![opencpn-2-27-2015 log](https://cloud.githubusercontent.com/assets/2363959/6423676/a07db7fa-beb7-11e4-8ab7-c12a72a79a8e.png)
you will have to take the png off.

**Comment by rgleason** [comment]
Dan and I have rewritten the New User Documentation with all new updated screenshots.
It should be pretty complete at this point.

See it here:
https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-new-OpenCPN-user

The screenshots are located on my virtual server, it is just much easier this way.

**Comment by rgleason** [comment]
Existing User Documentation. This needs to be re-wirtten with a different slant and with knowledge about the typical chart destination pathnames for the various OS

Windows:  C:\Users\All Users [or Username]\opencpn\Charts
Linux:  C:\Users\All Users [or Username]\opencpn\Charts   ???
Mac: C:\Users\All Users [or Username]\opencpn\Charts   ???

Any suggestions about what is useful and what is not?  You can leave notes or add comments...

See it here
https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-existing-OpenCPN-user

**Comment by rgleason** [comment]
The New users documentation is finished.  Review, comments & corrections would be appreciated.
https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-new-OpenCPN-user

**Comment by rgleason** [comment]
The Existing users documentation is finished. Review, comments & corrections would be appreciated.
https://github.com/rgleason/chartdldr_pi/wiki/User-Documentation-for-existing-OpenCPN-user

**Comment by rgleason** [comment]
Made more changes to Existing User Documentation and added
the Opencpn Documentation   https://github.com/rgleason/chartdldr_pi/wiki/Where-do-I-keep-my-charts
so that we can make the necessary changes to describe Chart Directories.

**Comment by rgleason** [comment]
Dan, Thank you for the improvements, both on the Plugin itself and the documentation!  You sure were able to sort out things!  Thank you from all of us users.  Also thanks to Pavel for his high standards and skills.

This morning I was thinking about possibly having a list of Error Messages in the documentation because this plugin is so dependent on outside connections, servers, file locations on servers, directories not changing and structure not changing, such as to an authentication or payment requirement. The user would need to know this information in order to make decisions about what to do.  I'm going to try to make a list, if such messages exist which I think they do:
1. Bad network connection.
2. The server did not respond or no longer exists.
3. The directory on the server does not exist.
4. The Catalog file does not exist, or cannot download the catalog.
5. The files the catalog refers to are not available?
6. Cannot download files, (maybe listing the files?)
7. The download of the files requires authentication or payment now.

Any others?  I think there are messages for most of these, but I have a pretty good Inet connection and did not unplug.

Also perhaps we should make a plea for users to advise of any freely downloadable ENC or RNC.

**Comment by transmitterdan** [comment]
The plugin doesn't know why a download failed. It just knows it failed. The wxCurl library code has several wx log debug messages that tell why a download fails but they only show up in the log file in the debug build as far as I know. If you run the debug build of the plugin under a debugger it writes quite a bit of info into the opencpn.log file. I never could get the debug build of the plugin to run under a nondebug OpenCPN executable in Windows. The only time I have seen any failure to download was when the local internet connection was down. The error dialog already tells the user to check that.

I suppose a web site could be down. It's easy to check that for yourself. If you use the catalog "Edit" button the URL that it will download is shown. Copy that URL to a web browser and see what happens. If you don't get anything that's your problem. Beyond that I don't think the plugin can help much with more error messages.

If a download requires authentication then the plugin doesn't handle that so far as I know right now. It could if we get that requirement.

We should stick with ENC and RNC files from official repositories in my opinion. We should not link to bootleg cache's of charts with unknown provinence.

**Comment by rgleason** [comment]
Ok I guess that addresses the idea of failure or error notices.

**Comment by nohal** [comment]
I have split the download and installation to a separate page and reorganized the basic usage instructions a bit to cover just the real basics and walk the user through the usage in a way that hopefully is logical. Any comments and corrections are welcome.
Will now move to the advanced instructions and hopefully release an RC tomorrow.

**Comment by transmitterdan** [comment]
Pavel,

I my opinion we should try to fix issue #63 before RC. I hated to find a bug this late but it is there so I think we have to fix it. I would appreciate your ideas on best way to fix.

**Comment by rgleason** [comment]
Pavel 
On my cell phone I see your first page edit but I am not finding the documentation pages or images. It may be my cell phone but a blank page comes up on the link. Did you move it somewhere?
Checked you wiki hoping it was there.

**Comment by rgleason** [comment]
Found it on bar to the riht. Links not working yet.

**Comment by rgleason** [comment]
Pavel,  
You and Dan edited down and improved the wiki documentation found on my github and I thought those were all good improvements.
I see the stable release 1.0 has documentation in it, however it was edited down further for size reasons since it is included in the plugin, and certain sections are missing.
Do you have any plan to include the full version in the Opencpn Manual under Plugins?
-- if so, perhaps we could include the more complete version with further edits?
--Dan replied to a user with a number of specific comments (earlier on the forum) and I think the documentation could respond to several of those comments.
--I would be willing to look at the comments and adjust the documentation where appropriate, but I do not know your plans at this point.

**Comment by nohal** [comment]
What exactly you think is missing?
- Installation instructions are kind of useless in the bundled docs, given the plugin is already installed and enabled for the user to be able to click on the Help button - If anywhere it should be integrated into the main OpenCPN documentation
- The "Where do I keep my charts" is covered in "Advanced" docs
  Anything else I stripped and you think should be included?
  I would really like to keep the documentation with the current Index + 2 pages if possible

**Comment by rgleason** [comment]
Pavel, 
1. I have acknowledged the need to keep the instructions short and compact for the files included with the plugin and have no problem with that or what you did.
2. However in my opinion they are very abbreviated must necessarily assume that users have some experience and knowledge, and drops details.
3. Therefore I am suggesting that the more complete version be uploaded to the Opencpn Plugin documentation.
4. Its your plugin, but we worked pretty hard to be clear and step by step about it. You're the boss, so I am not going to fight you about this, but I don't believe comprehension is improved by brevity over clarity. Additionally, the final edits in my repos were done by you and Dan and I thought those were good changes.

Bye

**Comment by nohal** [comment]
Rick...
I certainly appreciate improvements, but want to see that they actually improve something before they will be merged.
If you feel like I made something less clear than it was before, I'm sorry, but I feel the exact opposite.

**Comment by transmitterdan** [comment]
Lengthy documentation is seldom read in today's world. The plugin is more stable now and if new users just "take the defaults" they will usually be OK. If we see some queries raised more often we can address in the docs. For now let us see what queries users raise.

**Comment by rgleason** [comment]
Bye, doubt I'll be helping with documentation in the future.

**Comment by rgleason** [comment]
Do appreciate the plugin however.  Have been testing it out today, but its obvious that we have very different styles and I do not wish to waste my time anymore than is necessary.

**Comment by rgleason** [comment]
I was up too long and late last night, time zone problem, sorry.

Yes, nobody reads the docs until they have trouble.  "...When all else fails read the instructions." 
Yes, usually they ask forum questions first too.  If the instructions are too brief they aren't going to cover some of the questions IMHO

I added a single line in the main Opencpn docs to "Where do I keep my charts?

> Installation instructions are kind of useless in the bundled docs, given the plugin is already installed >.. to click on the Help button - If anywhere it should be integrated into the main OpenCPN >documentation

Agreed, when plugin is installed, however the "Requirements" section makes some important points which affect user experience should be read before install and after troubles. https://github.com/rgleason/chartdldr_pi/wiki/Installation     Perhaps this should be covered in the downloads section before they download the plugin or in the Main Opencpn Docs under "Plugins > ChartDownloader?

I will look at the Basics and Advanced pages and comment more later. 

I don't know why we can't have two versions - one for the Help button and a slightly longer one for the Web, but maybe it is better to have them the same.  

I do like the idea of having the help button right there with the plugin.

I also think you've brought the plugin many miles and everything works except Germany Inland DE_Inland  (more on that later). Right now I am testing the super update feature.

**Comment by nohal** [comment]
Rick...
We definitely can have two versions of the documentation. But the one maintained here is the one that gets bundled with the plugin and has to get the user going with as little reading and distraction from the main purpose as possible.

What is the problem with the German charts (apart from the fact that they can not be downloaded automatically so we tell the user what he has to do to get them)?

**Comment by rgleason** [comment]
> The plugin is more stable now and if new users just "take the defaults" they will usually be OK. 

Maybe you should advise new users to make a 'Charts' directory under their 'my docs' (or whatever) and not screw with the default settings until they know what they are doing?

**Comment by rgleason** [comment]
Pavel, that's great. 

I will review and modify the wiki on my repos, to get it smaller and briefer for the purpose of the Opencpn manual, and I will start by looking at the Help docs and just adding what I think might be needed. Ok?

Re German and Netherlands Inland, I just sent an email to you and Dan... don't know if it needs to be made an Issue...

Regarding the German charts.. from the browser download, then I made a new folder My Documents\Charts\ENC\DE_INLAND.   Moved all downloaded files into that folder and used Jzip to 'extract here'  which created folders like  
C:\Users\Frederick\Documents\OpenCPN-Data\Charts\ENC\DE_INLAND\ESK\ENC_ROOT
which was fine once the Chart Database was scanned.
